From 0c80bde138150fc7f5a275b075995ad8ba11caa9 Mon Sep 17 00:00:00 2001
From: Jarno Rajahalme <jarno@covalent.io>
Date: Fri, 15 May 2020 17:33:01 -0700
Subject: [PATCH] build: Avoid using git if not in a git repo

Do not use git if not in a git repo.

Only create GIT_VERSION if the existing file is already not the
same. This helps docker caching.

Store the list bpf files to a temporary file BPF_SRCFILES, which is
ignored by git like GIT_VERSION. This allows builds to succeed without
git.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
---
 .gitignore    |  1 +
 Makefile      |  9 ++++++---
 Makefile.defs | 11 ++++++++---
 3 files changed, 15 insertions(+), 6 deletions(-)

--- a/Makefile.defs
+++ b/Makefile.defs
@@ -38,7 +38,7 @@
 
 BPF_FILES_EVAL := $(shell git ls-files $(ROOT_DIR)/bpf/ | grep -v .gitignore | tr "\n" ' ')
 BPF_FILES ?= $(BPF_FILES_EVAL)
-BPF_SRCFILES := $(subst ../,,$(BPF_FILES))
+BPF_SRCFILES = $(shell cat $(ROOT_DIR)/BPF_SRCFILES)
 
 CILIUM_DATAPATH_SHA=$(shell cat $(BPF_FILES) | sha1sum | awk '{print $$1}')
 GOLDFLAGS += -X "github.com/cilium/cilium/pkg/datapath/loader.DatapathSHA=$(CILIUM_DATAPATH_SHA)"
